src/Microsoft.AspNet.JsonPatch/Adapters/SimpleObjectAdapter.cs

```diff
@@ -110,47 +119,52 @@ private void Add(string path, object value, T objectToApplyTo, Operation<T> oper
                 }
             }
 
-            var pathProperty = PropertyHelpers
-                .FindProperty(objectToApplyTo, actualPathToProperty);
-
+            var propertyMetadata = PropertyHelpers
+                .FindPropertyAndParent(objectToApplyTo, actualPathToProperty, ContractResolver);
 
             // does property at path exist?
-            if (pathProperty == null)
+            if (propertyMetadata == null)
             {
                 throw new JsonPatchException<T>(operationToReport,
                     string.Format("Patch failed: property at location path: {0} does not exist", path),
                     objectToApplyTo);
             }
+            if (propertyMetadata.Property.Ignored)
+            {
+                throw new JsonPatchException<T>(operationToReport,
+                    string.Format("Patch failed: cannot update property at location path: {0}", path),
```

Comment:
All of these strings should be moved to resources and reviewed. Please open a follow up item for that.